src/ToolInfo.php

```diff
@@ -26,67 +26,89 @@
     const COMPOSER_INSTALLED_FILE = '/../../composer/installed.json';
     const COMPOSER_PACKAGE_NAME = 'friendsofphp/php-cs-fixer';
 
-    public static function getComposerVersion()
+    /**
+     * @var string
+     */
+    private $script;
+
+    /**
+     * @var null|string
+     */
+    private $composerVersion;
+
+    /**
+     * @var null|bool
+     */
+    private $isInstalledAsPhar;
+
+    /**
+     * @var null|bool
+     */
+    private $isInstalledByComposer;
+
+    /**
+     * @var null|string
+     */
+    private $scriptDir;
+
+    public function __construct($script = null)
```

Comment:
So the code that uses `ToolInfo` should specify everytime `$_SERVER['SCRIPT_NAME']`?
Morover, it's a bit early to use 7.1-only feature, isn't it?